agent/main.go

```diff
@@ -90,6 +92,33 @@ func setupflags() {
 	for _, hiddenFlag := range hiddenFlags {
 		_ = pflag.CommandLine.MarkHidden(hiddenFlag)
 	}
+	feature.MutableGates.AddFlag(pflag.CommandLine)
+}
+
+func handleHostRegistration(k8sClient client.Client, hostName string, logger logr.Logger) (err error) {
+	if feature.Gates.Enabled(feature.SecureAccess) {
```

Comment:
what about setting as an env variable? Something like `SECURE_ACCESS=true`

Reply:
So in the case of feature-gate implementations for CAPI, I think they're just using the environment variables to set the flag, something like this:
`--feature-gates=ClusterTopology=${CLUSTER_TOPOLOGY:=false},KubeadmBootstrapFormatIgnition=${EXP_KUBEADM_BOOTSTRAP_FORMAT_IGNITION:=false}` Source: [Cluster Topology feature flag](https://github.com/kubernetes-sigs/cluster-api/blob/5756be13affef606b6683c2c413f0987c1f0e451/controlplane/kubeadm/config/manager/manager.yaml#L24)

I don't think we need to do anything in the code to handle this.

Reply:
That is if the env variables are passed as args via the `manager.yaml`
For host agent, we don't have anything like the `manager.yaml` and we start the binary directly.

What would happen if the user set the env variable and installed the byoh provider? 
1. the host agent is unaware of any such variable (the controllers on mgmt cluster would enable SecureAccess and agent controllers wouldn't know about it)
2. these variables need to be set on the host itself

@dharmjit Thinking about it, setting env variable on each host is tedious. Maybe the info should flow from the cluster template. But for this PR, let's assume the env variable needs to be set on the host so that we don't do any api / controller change for now.

Thoughts? @NilanjanDaw

Reply:
Okay realising we cannot rely on info from cluster template. This step is before applying the cluster template.

Reply:
Yeah, this makes sense. The host-agent requirement is unique in this regard compared to the ones in CAPI.

Reply:
As discussed, we will hold off on adding support for environment variables in a future PR.